erpnext/accounts/report/financial_statements.py

```diff
@@ -320,6 +315,22 @@ def set_gl_entries_by_account(company, from_date, to_date, root_lft, root_rgt, g
 
 	return gl_entries_by_account
 
+def get_additional_conditions(from_date, ignore_closing_entries, filters):
+	additional_conditions = []
+
+	if ignore_closing_entries:
+		additional_conditions.append("ifnull(voucher_type, '')!='Period Closing Voucher'")
+
+	if from_date:
+		additional_conditions.append("posting_date >= %(from_date)s")
+
+	if filters:
+		for key in filters:
```

Comment:
Why looping through all the filters? Instead you can loop only for cost center and project.

```
for key in ['cost_center', 'project']:
    if filters.get(key):
         // logic
```